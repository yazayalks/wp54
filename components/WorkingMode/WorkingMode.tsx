import styles from "../../styles/workingMode.module.scss";
import React, {FC} from "react";


const WorkingMode: FC = () => {
    return (
        <div data-testid="working_mode" className={styles.working_mode__inner}>
            <p data-testid="paragraph" className="small-text"><strong>Режим работы:</strong></p>
            <p data-testid="paragraph" className="small-text">Круглосуточно</p>
            <a data-testid="paragraph" href="tel:+7 (913) 381-22-81" className="small-text" title="Номер телефона лофта +7 (913) 381-22-81"><strong>+7 (913) 381-22-81</strong></a>
            <p data-testid="paragraph" className="small-text">Ул. Ватутина, 19</p>
        </div>
    );
};

export default WorkingMode;