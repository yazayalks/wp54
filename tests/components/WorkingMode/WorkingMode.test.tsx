import React from "react";
import "@testing-library/react";
import {render, screen} from "@testing-library/react";
import WorkingMode from "@components/WorkingMode/WorkingMode";

describe('Компонент WorkingMode', () => {

    beforeEach(() => {
        render(<WorkingMode/>);
    });
    test('Рендер компонента WorkingMode', () => {

    });
    test('Наличие класса у  компонента WorkingMode', () => {
        const element = screen.getByTestId("working_mode");
        expect(element).toHaveClass("working_mode__inner");
    });
    test('Наличие 4 параграфов', () => {
        const paragraphs = screen.getAllByTestId("paragraph");
        expect(paragraphs.length).toBe(4)
    });
    test('Снэпшот WorkingMode', () => {
        const element = screen.getByTestId("working_mode");
        expect(element).toMatchSnapshot();
    });
});