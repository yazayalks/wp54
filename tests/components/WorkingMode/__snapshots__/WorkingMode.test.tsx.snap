// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент WorkingMode Снэпшот WorkingMode 1`] = `
<div
  class="working_mode__inner"
  data-testid="working_mode"
>
  <p
    class="small-text"
    data-testid="paragraph"
  >
    <strong>
      Режим работы:
    </strong>
  </p>
  <p
    class="small-text"
    data-testid="paragraph"
  >
    Круглосуточно
  </p>
  <a
    class="small-text"
    data-testid="paragraph"
    href="tel:+7 (913) 381-22-81"
  >
    <strong>
      +7 (913) 381-22-81
    </strong>
  </a>
  <p
    class="small-text"
    data-testid="paragraph"
  >
    Ул. Ватутина, 19
  </p>
</div>
`;
